Домашнее задание 11

KrisM1307 opened the pull request:
Очень запуталась, выполняя 2 уровня сложности, таймер неправильно работает и не считаются ошибки и неправильные слова(

KrisM1307 committed:
Домашнее задание 11

Очень запуталась, выполняя 2 уровня сложности, таймер неправильно работает и не считаются ошибки и неправильные слова(

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,100 @@
+document.addEventListener('keydown', event => {
+    const typedLetter = event.key;
+
+    if (currentIndex === 0) {
```
Кристина,  тут есть нюанс. Обрати внимание,  что с каждым словом таймер будет ускоряться. Такого не должно быть. Этой проверки недостаточно. 

Лучше введи переменную,  например isStarted, которая первоначально false. Тут делай на неё проверку,  если она false, то запускай таймер , а её меняй на true. Тогда при новом слове,  сюда ко не зайдет
А уже в resetGame опять пtрезаписывай её на false

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,100 @@
+
+    if (typedLetter === currentWord[currentIndex]) {
+        const spans = wordElement.querySelectorAll('span');
+        spans[currentIndex].classList.add('c');
```
Если буква красная,  и её повторно правильно ввели,  то она не становится зеленой)  Потому что на ней весит класс w, а он в css ниже чем c и с его не перебивает. Подумай как можно поправить)

KrisM1307 committed:
Добавила новую проверку таймера и исправила ошибку с добавлением класса

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,104 @@
+        wordElement.append(span);
+    }
+
+    updateMistakesCounter();
```
Нет смысла прямо отдельную функцию для одной строки ,  просто сразу пиши 
mistakesElement.textContent = mistakesInCurrentWord;


а вот это удаляем
function updateMistakesCounter() {
    mistakesElement.textContent = mistakesInCurrentWord;
}

и тут же вызываем функцию checkEndGame();

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,104 @@
+    updateMistakesCounter();
+}
+
+function updateMistakesCounter() {
```
Это удаляем

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,104 @@
+            correctWords++;
+            correctCountElement.textContent = correctWords;
+            updateWord();
+            if (correctWords === 5) {
```
79-82 удаляем ,  проверка уже в функции updateWord

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,104 @@
+        if (currentIndex === currentWord.length) {
+            correctWords++;
+            correctCountElement.textContent = correctWords;
+            updateWord();
```
Чтобы у нас последняя буква слова успевала окрасится,  нужно использовать setTimeout
setTimeout(updateWord, 0);

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,104 @@
+            }
+        }
+    } else {
+        mistakesInCurrentWord++;
```
Тут же увеличиваем allMistakes. Если сделали 2 ошибки в слове,  то считается,  что уже 2 слова ввели с ошибками, иначе всегда будет только победа

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,104 @@
+
+const wordElement = document.querySelector('.word');
+const correctCountElement = document.querySelector('.correct-count');
+const mistakesElement = document.querySelector('.word-mistakes');
```
Делаем выборку контейнера с классом .wrong-count

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,104 @@
+        }
+    } else {
+        mistakesInCurrentWord++;
+        mistakesElement.textContent = mistakesInCurrentWord;
```
Вот тут еще отображаем allMistakes, выборку контейнера для этого вначале сделаешь и по такому же принципу
выборка контейнера.textContent = allMistakes;

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,104 @@
+    clearInterval(timerInterval);
+    isStarted = false;
+    correctWords = 0;
+    mistakesInCurrentWord = 0;
```
Обнуляешь allMistakes

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,104 @@
+    elapsedTime = 0;
+    updateWord();
+    correctCountElement.textContent = correctWords;
+    mistakesElement.textContent = mistakesInCurrentWord;
```
Выводишь все ошибки по такому же принципу
выборка контейнера.textContent = allMistakes;

KrisM1307 committed:
Исправления)

Еще исправила, чтобы максимальное количество слов было 5, включая правильные и неправильные
И у меня еще каждая ошибка в одном слове считалась как несколько неправильных слов, вроде бы тоже исправила)

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,120 @@
+}
+
+function checkEndGame() {
+    if (completedWords === 5) {
```
Не совсем понимаю, почему всего введенных слов может быть только 5?
![Screenshot_168](https://github.com/user-attachments/assets/efb7a4c4-d49a-4bbe-beb7-087ed2639b9b)
Почему закончилась игра?

mdubik777 commented on script.js:
```diff
@@ -0,0 +1,120 @@
+            }
+            correctCountElement.textContent = correctWords;
+            completedWords++;
+            updateWord();
```
вот конец игры и последняя буква не зеленая. Почему не добавила setTimeout?
![Screenshot_168](https://github.com/user-attachments/assets/a6b454cd-540c-4f76-a1f1-6b50f6f1611e)
И счетчик правильных ответов не обновился

KrisM1307 committed:
Убрала максимальное количество слов и добавила setTimeout
